byron/ledger/executable-spec/test/Ledger/Delegation/Properties.hs

```diff
@@ -269,8 +274,12 @@ instance HasTrace DBLOCK where
         n <- Gen.integral (Range.linear 0 13)
         pure $! Set.fromAscList $ mkVKeyGenesis <$> [0 .. n]
 
-  sigGen (Just InvalidDelegationCertificate) _ (env, _st) =
-    DBlock <$> nextSlotGen env <*> Gen.list (Range.constant 0 10) randomDCertGen
+  sigGen maybePredFailure _ (env, st) =
+    case maybePredFailure of
+      Just pf | toConstr (DPF undefined) == pf ->
+        DBlock <$> nextSlotGen env <*> Gen.list (Range.constant 0 10) randomDCertGen
+      _ ->
+        DBlock <$> nextSlotGen env <*> sigGen @DELEG Nothing env st
```

Comment:
I mentioned this on Slack but wanted to repeat it here. In my goblins work I added an equality predicate to the `STS` typeclass, because I realized that `toConstr` only checked equality "1 level deep" in terms of constructors. For example:
```
> toConstr (Just True) == toConstr (Just False)
True
> toConstr (Just True) == toConstr (Nothing :: Maybe Bool)
False
```
So we see proper equality checks at the `Maybe` level, but not at the enclosed `Bool` level. That's why I wrote a custom equality predicate:
https://github.com/input-output-hk/cardano-ledger-specs/commit/0391658ae9d7a0c57002d0115d92969d329c3c62
https://github.com/input-output-hk/cardano-ledger-specs/blob/446cde960be921943e7ab5f97e357ab18e8407bf/byron/ledger/executable-spec/src/Ledger/Delegation.hs#L277

Reply:
> I mentioned this on Slack but wanted to repeat it here.

Yes! Repetition is good :) I made a comment about this here https://github.com/input-output-hk/cardano-ledger-specs/commit/0391658ae9d7a0c57002d0115d92969d329c3c62#r34630580 Repetition is good ;)